impeller/typographer/backends/skia/text_render_context_skia.h

```diff
@@ -6,9 +6,37 @@
 
 #include "flutter/fml/macros.h"
 #include "impeller/typographer/text_render_context.h"
+#include "third_party/skia/include/core/SkBitmap.h"
 
 namespace impeller {
 
+class DeviceBuffer;
+class Allocator;
+
+/// @brief An implementation of an SkBitmap allocator that deferrs allocation to
+///        an impeller allocator. This allows usage of Skia software rendering
+///        to write to a host buffer or linear texture without an extra copy.
+///
+///        This class is an exact copy of the implementation in
+///        image_decode_impeller.cc due to the lack of a reasonable library
+///        that could be shared.
```

Comment:
Right now we have `display_list:skia_conversions`, `aiks`, and `image` and this that know about Skia.

It seems like this could probably go into `image`, and we could add a dep to `impeller/image` to both `lib/ui` and `impeller/typographer`. Doesn't have to be done in this patch but would be good as a follow up if not done here.